yarn-project/aztec-node/src/aztec-node/server.test.ts

```diff
@@ -0,0 +1,220 @@
+
+    const store = mock<AztecLmdbStore>();
+
+    const aztecNodeConfig: AztecNodeConfig = getConfigEnvVars();
+
+    node = new AztecNodeService(
+      {
+        ...aztecNodeConfig,
+        l1Contracts: {
+          ...aztecNodeConfig.l1Contracts,
+          rollupAddress: EthAddress.ZERO,
+          registryAddress: EthAddress.ZERO,
+          inboxAddress: EthAddress.ZERO,
+          outboxAddress: EthAddress.ZERO,
+          availabilityOracleAddress: EthAddress.ZERO,
+        },
+      },
+      p2p,
+      l2BlockSource,
+      l2LogsSource,
+      l2LogsSource,
+      contractSource,
+      l1ToL2MessageSource,
+      worldState,
+      undefined,
+      31337,
+      1,
+      globalVariablesBuilder,
+      store,
+      new TestCircuitVerifier(),
+      new NoopTelemetryClient(),
+    );
+  });
+
+  describe('tx validation', () => {
+    it('tests that the node correctly validates double spends', async () => {
+      const txs = [mockTxForRollup(0x10000), mockTxForRollup(0x20000)];
+      txs.forEach(tx => {
+        tx.data.constants.txContext.chainId = chainId;
+      });
+      const doubleSpendTx = txs[0];
+      const doubleSpendWithExistingTx = txs[1];
+
+      const mockedGlobalVariables = new GlobalVariables(
+        chainId,
+        version,
+        new Fr(lastBlockNumber + 1),
+        new Fr(1),
+        Fr.ZERO,
+        coinbase,
+        feeRecipient,
+        gasFees,
+      );
+
+      globalVariablesBuilder.buildGlobalVariables
+        .mockResolvedValueOnce(mockedGlobalVariables)
+        .mockResolvedValueOnce(mockedGlobalVariables);
+
+      expect(await node.isValidTx(doubleSpendTx)).toBe(true);
+
+      // We push a duplicate nullifier that was created in the same transaction
+      doubleSpendTx.data.forRollup!.end.nullifiers.push(doubleSpendTx.data.forRollup!.end.nullifiers[0]);
+
+      expect(await node.isValidTx(doubleSpendTx)).toBe(false);
+
+      globalVariablesBuilder.buildGlobalVariables
+        .mockResolvedValueOnce(mockedGlobalVariables)
+        .mockResolvedValueOnce(mockedGlobalVariables);
+
+      expect(await node.isValidTx(doubleSpendWithExistingTx)).toBe(true);
+
+      // We make a nullifier from `doubleSpendWithExistingTx` a part of the nullifier tree, so it gets rejected as double spend
+      const doubleSpendNullifier = doubleSpendWithExistingTx.data.forRollup!.end.nullifiers[0].toBuffer();
+      merkleTreeOps.findLeafIndex.mockImplementation((treeId: MerkleTreeId, value: any) => {
+        return Promise.resolve(
+          treeId === MerkleTreeId.NULLIFIER_TREE && value.equals(doubleSpendNullifier) ? 1n : undefined,
+        );
+      });
+
+      expect(await node.isValidTx(doubleSpendWithExistingTx)).toBe(false);
+    });
+
+    it('tests that the node correctly validates chain id', async () => {
+      const tx = mockTxForRollup(0x10000);
+      tx.data.constants.txContext.chainId = chainId;
+
+      const mockedGlobalVariables = new GlobalVariables(
```

Comment:
I'm surprised by all of this, did we have no tests for the node validations at all?